[docs] Correct grammatically incorrect sentences in CONTRIBUTING.md (#34949)
mui/material-ui: CONTRIBUTING.md

### How to increase the chance of being accepted?

Continuous Integration (CI) runs a series of checks automatically when a Pull Request is opened. If you're
unsure if your changes will pass, you can always open a Pull Request, and the GitHub UI will display a summary
of the results. If any of them fail, refer to [Checks and how to fix them](#checks-and-how-to-fix-them).

Make sure the following is true:

<!-- #default-branch-switch -->

> - The branch is targeted at `master` for ongoing development. With all tests passing, we do our best to keep `master` in good shape. Code that lands in `master` must be compatible with the latest stable release. It may contain additional features but no breaking changes. We should be able to release a new minor version from the tip of `master` at any time.

Reviewer: I think this better conveys the intent of the original text, while cutting out some of the fluff:

```suggestion
- The branch is targeted at `master` for ongoing development. All tests are passing. Code that lands in `master` must be compatible with the latest stable release. It may contain additional features but no breaking changes. We should be able to release a new minor version from the tip of `master` at any time.
```

- The demos are up-to-date. If demos were changed, make sure `yarn docs:typescript:formatted` does not introduce changes. See [about writing demos](#3-write-the-content-of-the-demo).
- The Pull Request title follows the pattern `[Component] Imperative commit message`. (See: [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/) for a great explanation).

If you have missed a step, don't worry, the Continuous Integration will run a thorough test on your commits, and the maintainers of the project can assist.

If your pull request addresses an open issue, make sure to link the PR to that issue.
Use any [supported GitHub keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) in the PR description to automatically link them.

> It makes it easier to understand where the PR is coming from and speeds things up as the issue gets closed when the PR is merged.

Reviewer: I would keep the original text here, but just add a comma.

```suggestion
This makes it easier to understand where the PR is coming from, and also speeds things up as the issue gets closed when the PR is merged.
```
